vtysh/bgp_vty.c

```diff
@@ -3228,6 +3249,103 @@ DEFUN(no_neighbor_peer_group_remote_as,
     return CMD_SUCCESS;
 }
 
+static int
+cli_neighbor_local_as_cmd_execute(char *vrf_name, const char *argv[])
+{
+    const char *peer_str = argv[0];
+    int64_t local_as = strtoll(argv[1], NULL, 10);
+    const struct ovsrec_bgp_router *bgp_router_context;
+    const struct ovsrec_vrf *vrf_row;
+    const struct ovsrec_bgp_neighbor *ovs_bgp_neighbor, *ovs_peer_grp;
+    struct ovsdb_idl_txn *txn;
+    int64_t asn = (int64_t)vty->index;
+
+    START_DB_TXN(txn);
+
+    vrf_row = get_ovsrec_vrf_with_name(vrf_name);
+    if (!vrf_row) {
+        ERRONEOUS_DB_TXN(txn, BGP_ERR_NO_VRF_FOUND);
+    }
+
+    bgp_router_context = get_ovsrec_bgp_router_with_asn(vrf_row, asn);
+    if (!bgp_router_context) {
+        ERRONEOUS_DB_TXN(txn, BGP_ERR_ROUTER_IS_NOT_CONFIGURED);
+    }
+
+    ovs_bgp_neighbor =
+    get_bgp_peer_group_with_bgp_router_and_name(bgp_router_context, peer_str);
+    if (!ovs_bgp_neighbor) {
+        ABORT_DB_TXN(txn, BGP_ERR_NEIGHBOR_IS_NOT_CONFIGURED);
+    }
+
+    if(ovs_bgp_neighbor->bgp_peer_group) {
+        ABORT_DB_TXN(txn, BGP_ERR_INVALID_FOR_PEER_GROUP_MEMBER);
+    }
+
+    if (is_ibgp_peer(ovs_bgp_neighbor, asn)) {
+        ABORT_DB_TXN(txn, BGP_ERR_LOCAL_AS_ALLOWED_ONLY_FOR_EBGP);
+    }
+
+    if (local_as == asn) {
+        ABORT_DB_TXN(txn, BGP_ERR_CANNOT_HAVE_LOCAL_AS_SAME_AS);
+    }
+
+    ovsrec_bgp_neighbor_set_local_as(ovs_bgp_neighbor, &local_as, 1);
+
+    if (object_is_peer_group(ovs_bgp_neighbor)) {
+        ovs_peer_grp = ovs_bgp_neighbor;
+        OVSREC_BGP_NEIGHBOR_FOR_EACH(ovs_bgp_neighbor, idl) {
+            if (is_member_of_peer_group(ovs_bgp_neighbor, ovs_peer_grp) && !is_ibgp_peer(ovs_bgp_neighbor, asn)) {
+                ovsrec_bgp_neighbor_set_local_as(ovs_bgp_neighbor, &local_as, 1);
+            }
+        }
+    }
+
+    END_DB_TXN(txn);
+}
+
+static int
+cli_no_neighbor_local_as_cmd_execute(char *vrf_name, const char *argv[])
+{
+    const char *peer_str = argv[0];
+    const struct ovsrec_bgp_router *bgp_router_context;
+    const struct ovsrec_vrf *vrf_row;
+    const struct ovsrec_bgp_neighbor *ovs_bgp_neighbor, *ovs_peer_grp;
+    struct ovsdb_idl_txn *txn;
+    int64_t asn = (int64_t)vty->index;
+
+    START_DB_TXN(txn);
+
+    vrf_row = get_ovsrec_vrf_with_name(vrf_name);
+    if (!vrf_row) {
+        ERRONEOUS_DB_TXN(txn, BGP_ERR_NO_VRF_FOUND);
+    }
+
+    bgp_router_context = get_ovsrec_bgp_router_with_asn(vrf_row, asn);
+    if (!bgp_router_context) {
+        ERRONEOUS_DB_TXN(txn, BGP_ERR_ROUTER_IS_NOT_CONFIGURED);
+    }
+
+    ovs_bgp_neighbor =
+    get_bgp_peer_group_with_bgp_router_and_name(bgp_router_context, peer_str);
+    if (!ovs_bgp_neighbor) {
+        ABORT_DB_TXN(txn, BGP_ERR_NEIGHBOR_IS_NOT_CONFIGURED);
+    }
+
```

Comment:
Missing check for peer-group membership?
` if(ovs_bgp_neighbor->bgp_peer_group) {`
 `ABORT_DB_TXN(txn, BGP_ERR_INVALID_FOR_PEER_GROUP_MEMBER);}`